src/plugins/ui_core_views/row_size.ts

```diff
@@ -0,0 +1,164 @@
+import { DEFAULT_CELL_HEIGHT } from "../../constants";
+import {
+  deepCopy,
+  getAddHeaderStartIndex,
+  getDefaultCellHeight,
+  insertItemsAtIndex,
+  range,
+  removeIndexesFromArray,
+} from "../../helpers";
+import { Command } from "../../types";
+import { CellPosition, HeaderIndex, Immutable, Pixel, UID } from "../../types/misc";
+import { UIPlugin } from "../ui_plugin";
+
+interface HeaderSizeState {
+  sizes: Immutable<Record<UID, Array<Pixel>>>;
+}
+
+export class UIRowSizePlugin extends UIPlugin<HeaderSizeState> implements HeaderSizeState {
+  static getters = ["getRowSize"] as const;
+
+  readonly sizes: Immutable<Record<UID, Array<Pixel>>> = {};
+
+  private ctx = document.createElement("canvas").getContext("2d")!;
+
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "START":
+        for (const sheetId of this.getters.getSheetIds()) {
+          this.history.update("sizes", sheetId, this.computeRowSizes(sheetId));
+        }
+        break;
+      case "CREATE_SHEET": {
+        this.history.update("sizes", cmd.sheetId, this.computeRowSizes(cmd.sheetId));
+        break;
+      }
+      case "DUPLICATE_SHEET": {
+        const sizes = deepCopy(this.sizes[cmd.sheetId]);
+        this.history.update("sizes", cmd.sheetIdTo, sizes);
+        break;
+      }
+      case "DELETE_SHEET":
+        const sizes = { ...this.sizes };
+        delete sizes[cmd.sheetId];
+        this.history.update("sizes", sizes);
+        break;
+      case "REMOVE_COLUMNS_ROWS": {
+        if (cmd.dimension === "COL") return;
+        let sizes = [...this.sizes[cmd.sheetId]];
+        sizes = removeIndexesFromArray(sizes, cmd.elements);
+        this.history.update("sizes", cmd.sheetId, sizes);
+        break;
+      }
+      case "ADD_COLUMNS_ROWS": {
+        if (cmd.dimension === "COL") {
+          return;
+        }
+        const oldSizes = [...this.sizes[cmd.sheetId]];
+        const addIndex = getAddHeaderStartIndex(cmd.position, cmd.base);
+        const newRows = Array(cmd.quantity).fill(DEFAULT_CELL_HEIGHT);
+        const newSizes = insertItemsAtIndex(oldSizes, newRows, addIndex);
+        this.history.update("sizes", cmd.sheetId, newSizes);
+        break;
+      }
+      case "RESIZE_COLUMNS_ROWS":
+        {
+          const sheetId = cmd.sheetId;
+          if (cmd.dimension === "ROW") {
+            for (let el of cmd.elements) {
+              this.history.update("sizes", sheetId, el, this.getRowComputedSize(sheetId, el));
+            }
+          } else {
+            // Recompute row heights on col size change, they might have changed because of wrapped text
+            for (let row of range(0, this.getters.getNumberRows(sheetId))) {
+              this.history.update("sizes", sheetId, row, this.getRowComputedSize(sheetId, row));
+            }
+          }
+        }
+        break;
+      case "UPDATE_CELL":
```

Comment:
isn't this a perf killer ? If I update every cell of a row (let's say a large copy-paste) , the size is recomputed every time.